internal/typeinfo/typeinfo_test.go

```diff
@@ -0,0 +1,227 @@
+package typeinfo
+
+import (
+	"reflect"
+	"testing"
+
+	. "gopkg.in/check.v1"
+)
+
+var _ arg = &structInfo{}
+var _ arg = &mapInfo{}
+
+func TestTypeInfo(t *testing.T) { TestingT(t) }
+
+type typeInfoSuite struct{}
+
+var _ = Suite(&typeInfoSuite{})
+
+func (s *typeInfoSuite) TestArgInfoStruct(c *C) {
+	type myStruct struct {
+		ID        int    `db:"id"`
+		Name      string `db:"name,omitempty"`
+		ValidTag1 int    `db:"_i_d_55_"`
+		ValidTag2 int    `db:"IdENT99"`
+		NotInDB   string
+	}
+
+	argInfo, err := GenerateArgInfo([]any{myStruct{}})
+	c.Assert(err, IsNil)
+
+	// The struct fields in this list are ordered according to how sort.Strings
+	// orders the tag names. This matches the order of the Output values from
+	// AllStructOutputs.
+	structFields := []struct {
+		fieldName string
+		index     int
+		omitEmpty bool
+		tag       string
+	}{{
+		fieldName: "ValidTag2",
+		index:     3,
+		omitEmpty: false,
+		tag:       "IdENT99",
+	}, {
+		fieldName: "ValidTag1",
+		index:     2,
+		omitEmpty: false,
+		tag:       "_i_d_55_",
+	}, {
+		fieldName: "ID",
+		index:     0,
+		omitEmpty: false,
+		tag:       "id",
+	}, {
+		fieldName: "Name",
+		index:     1,
+		omitEmpty: true,
+		tag:       "name",
+	}}
+
+	allOutputs, names, err := argInfo.AllStructOutputs("myStruct")
+	c.Assert(err, IsNil)
+	c.Assert(allOutputs, HasLen, len(structFields))
+
+	for i, t := range structFields {
+		expectedStructField := &structField{
+			name:       t.fieldName,
+			structType: reflect.TypeOf(myStruct{}),
```

Comment:
[Nit]: You can define the type above instead of calling reflect for each subtest.

Reply:
Changed, but its just a test